### PR TITLE
Make geometry elements importable functions, and fill cells with materials

### LINKED ISSUE #8: Fuel element geometry does not show cladding on the outside, and appears larger than specified in schnitzler. Borehole position do not visually correspond to image in schnitzler
image

working on fixing

### PR DESCRIPTION
**Pull request by Edgar-21**
closes #8 & #2 or at least it should

Please take  a close look at how I changed the geometry of the cells, visually they now match Schnitzler better. Additionally please take a close look at which materials are assigned to which cells :)

### CODE REVIEW

File: geometry_definitions.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    tie_tube_assembly = openmc.model.HexagonalPrism(orientation='x',
+                                                    edge_length=inner_edge_length)
+    tie_tube_assembly_cladding = openmc.model.HexagonalPrism(orientation='x',
+                                                             edge_length=assembly_edge_length)
+
+    # OpenMC Cells and Universes
+    inner_hydrogen_cell = openmc.Cell(region=-inner_hydrogen, fill=hydrogen_inner)
+    inner_tie_tube_cell = openmc.Cell(
+        region=+inner_hydrogen & - inner_tie_tube, fill=inconel)
+    first_gap_cell = openmc.Cell(region=+inner_tie_tube & -first_gap)
```
Comment (dean-krueger) [inline geometry_definitions.py:112]: All four gaps should be filled with hydrogen according to that table I sent

File: geometry_definitions.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    moderator_tube_cell = openmc.Cell(region=+first_gap & -moderator_tube,fill=ZrH)
+    outer_hydrogen_cell = openmc.Cell(region=+moderator_tube & -second_gap, fill=hydrogen_outer)
+    outer_tie_tube_cell = openmc.Cell(region=+second_gap & -outer_tie_tube, fill=inconel)
+    third_gap_cell = openmc.Cell(region=+outer_tie_tube & -third_gap)
+    insulator_cell = openmc.Cell(region=+outer_tie_tube & -insulator,fill = ZrC)
```
Comment (dean-krueger) [inline geometry_definitions.py:117]: I'm not entirely sure how to do this, but the table I sent (which you didn't have while you were making this) indicates that the Insulator Tube is made of ZrC at 50% TD (which I think means theoretical density). I don't know openmc materials well enough to know if we can do this, but maybe there's some density attribute we can modify?

File: geometry_definitions.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    fourth_gap_cell = openmc.Cell(region=+insulator & -fourth_gap)
+
+    # the fill on these two are also educated guesses, print statements are to ensure we fix this
+    tie_tube_assembly_cell = openmc.Cell(region=-tie_tube_assembly & +fourth_gap, fill=ZrH)
+    tie_tube_assembly_cladding_cell = openmc.Cell(
+        region=+tie_tube_assembly & -tie_tube_assembly_cladding, fill=ZrC)
```
Comment (dean-krueger) [inline geometry_definitions.py:123]: ZrC is correct here, and it should be 100% TD (which I say to differentiate it from the earlier comment about 50% TD ZrC)

File: geometry_definitions.py (hunk 2 above)
Comment (Edgar-21) [reply geometry_definitions.py:117]: it appears to be possible with `openmc.Material.mix_materials()`

File: geometry_definitions.py (hunk 1 above)
Comment (Edgar-21) [reply geometry_definitions.py:112]: filling all gaps with `hydrogen_outer`, doubtful those very thin layers will make much of an impact but we should explore that to be sure

File: geometry_definitions.py (hunk 2 above)
Comment (Edgar-21) [reply geometry_definitions.py:117]: I'll add it to the material definitions in pyne however, to keep all material definitions together

**Comment by Edgar-21** [comment]
image
tie tube by material now, thanks for that info

**Pull request merged**